kyo-data/shared/src/main/scala/kyo/Text.scala

```diff
@@ -0,0 +1,288 @@
+package kyo
+
+import Text.internal.*
+import scala.annotation.tailrec
+
+opaque type Text >: String = String | Op
```

Comment:
Note how `Text` is marked as a super class of `String`, which enables passing strings where `Text` is expected.